src/python/pants/goal/migrate_call_by_name.py

```diff
@@ -556,20 +567,89 @@ def map_dict_form_get_to_new_syntax(
 # ------------------------------------------------------------------------------------------
 
 
-def remove_unused_implicitly(call: cst.Call, called_func: cst.FunctionDef) -> cst.Call:
+def fix_implicitly_usage(call: cst.Call, target_func: cst.FunctionDef) -> cst.Call:
```

Comment:
Reviewers can largely ignore this function - it's getting updated, modified, and re-factored basically every new migration to handle missing `implicitly` cases - and once those cases are stable and tested, I will re-factor this and also extract it to a separate fixer.

Only committing here in the event anyone else wants to run migration, this version better handles implicitly cleanup.

The tests in `migrate_call_by_name_tests.py` are likely more interesting.